src/feed/api.ts

```diff
@@ -64,23 +72,51 @@ export async function writeFeedData(
  * @param connection connection information for data uploading
  * @param topic key for data
  * @param data data to upload
- * @param privateKey private key to sign data
+ * @param wallet feed owner's wallet
  * @param epoch feed epoch
  */
 export async function writeFeedDataRaw(
   connection: Connection,
   topic: string,
   data: Uint8Array,
-  privateKey: string | Uint8Array,
+  wallet: utils.HDNode,
```

Comment:
almost everywhere inside the project, HDNode is used, since it is created from seed. Wallet is used in deprecated methods, which should be removed soon. Does it really make sense to switch to Wallet?

Reply:
you can also do both since you pass `wallet.privateKey` to the socWriter, so it can be `utils.HDNode | Wallet`